render/render.js

```diff
@@ -597,6 +604,9 @@ module.exports = function($window) {
 	}
 
 	function render(dom, vnodes) {
+		// granular update through render(vnode)
+		if (dom.tag && dom.dom.vnodes) render(dom.dom, Object.assign({}, dom))
```

Comment:
Did you forget a `return` here?

If so, just do this to cut the object overhead:

1. Change the above lines to this:

    ```js
    var force = false

    if (dom.tag != null && dom.dom.vnodes != null) {
        force = true
        vnodes = [dom]
         dom = dom.dom
    }
    ```

2. Add a new parameter to [`updateNodes()`](https://github.com/ludbek/mithril.js/blob/eed2500c3db2c0db918f0c727246cd7d0ed40bf8/render/render.js#L147) for that `force` option, set to `force` in the call to it later within this function.

If it was intentionally omitted, why are you double-rendering the vnode, when the second one is guaranteed to be redundant?